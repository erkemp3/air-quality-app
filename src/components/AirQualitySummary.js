/* eslint-disable */

import React from "react";
import PropTypes from "prop-types";
import "../styles/AirQualitySummary.scss";
import ClearIcon from "@material-ui/icons/Clear";

const AirQualitySummary = ({ onSubmit, summaries }) => {
  const dateFormatter = (string) => {
    const one = string.split("").splice(8, 2).join("");
    const two = string.split("").splice(5, 2).join("");
    const three = string.split("").splice(0, 4).join("");
    const secondHalf = string.split("").splice(11, 5).join("");
    return `${one}-${two}-${three} - ${secondHalf}`;
  };

  const measurementsFormatter = (obj) => {
    let string = "";
    for (let i = 0; i < obj.measurements.length - 1; i++) {
      string += `${obj.measurements[i].parameter.toUpperCase()}: ${
        obj.measurements[i].value
      }, `;
    }
    return `${string} ${obj.measurements[
      obj.measurements.length - 1
    ].parameter.toUpperCase()}: ${
      obj.measurements[obj.measurements.length - 1].value
    }`;
  };

  return (
    <div className="summaries">
      {summaries.map((summary) => (
        <div className="summary" key={summary.location}>
          <div className="button-div">
            <button
              type="submit"
              onClick={() => {
                onSubmit(summary.location);
              }}
              className="delete-button"
            >
              <ClearIcon />
            </button>
          </div>
          <p className="date">
            LAST UPDATED: {dateFormatter(summary.measurements[0].lastUpdated)}
          </p>
          <p className="location">{summary.location}</p>
          <p className="city">in {summary.city}, United Kingdom</p>
          <p className="values">Values: {measurementsFormatter(summary)}</p>
        </div>
      ))}
    </div>
  );
};

AirQualitySummary.propTypes = {
  summaries: PropTypes.arrayOf(
    PropTypes.shape({
      city: PropTypes.string.isRequired,
      location: PropTypes.string.isRequired,
      measurements: PropTypes.arrayOf(
        PropTypes.shape({
          lastUpdated: PropTypes.string.isRequired,
          parameter: PropTypes.string.isRequired,
          value: PropTypes.number.isRequired,
        }).isRequired
      ).isRequired,
    }).isRequired
  ).isRequired,
};

export default AirQualitySummary;
