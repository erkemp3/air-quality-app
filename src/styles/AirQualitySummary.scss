$base-purple: rgb(120, 55, 175);
$base-blue: rgba(68, 117, 175, 1);
$dark-grey: rgba(0, 0, 0, 0.815);
$light-grey: rgba(10, 10, 10, 0.054);
$medium-grey: rgba(10, 10, 10, 0.671);

.summaries {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  margin-top: 50px;
  justify-content: center;
  padding: 1rem;
}

.summary {
  display: flex;
  flex-direction: column;
  background-color: white;
  min-height: 175px;
  overflow: hidden;
  height: 1%;
  width: 420px;
  border-radius: 12px;
  margin: 20px;
  box-shadow: 0 15px 30px -10px rgba(0, 0, 0, 0.25);
  transition: all 0.25s ease-in-out;
}

.summary:hover {
  transform: scale(1.02);
}

.button-div {
  display: flex;
  justify-content: right;

  .delete-button {
    margin-top: 5px;
    margin-right: 5px;
    border: none;
    background-color: transparent;
    cursor: pointer;
  }
}

.date {
  text-align: left;
  margin-top: 0;
  margin-bottom: 7px;
  margin-left: 30px;
  font-size: 13px;
  font-weight: 500;
  color: $dark-grey;
}

.location {
  margin-top: 0;
  margin-left: 30px;
  margin-bottom: 7px;
  text-align: left;
  color: $base-purple;
  font-weight: 700;
  font-size: 22px;
}

.city {
  text-align: left;
  margin-top: 0;
  margin-bottom: 7px;
  margin-left: 30px;
  font-size: 16px;
  font-weight: 400;
  color: $medium-grey;
}

.values {
  text-align: left;
  margin-top: 0;
  margin-left: 30px;
  font-size: 15px;
  font-weight: 600;
  color: $dark-grey;
}

@media (max-width: 900px) {
  .summaries {
    margin-top: 15px;
  }
  .summary {
    overflow: hidden;
    height: 1%;
    width: 320px;
    min-height: 180px;
    margin-right: 20px;
    margin-left: 20px;
    margin-top: 0px;
    margin-bottom: 30px;
  }
  .date {
    margin-left: 17px;
    font-size: 12px;
  }
  .city {
    font-size: 14px;
    margin-left: 17px;
  }
  .values {
    font-size: 13px;
    margin-left: 17px;
    margin-right: 10px;
  }
  .location {
    font-size: 20px;
    margin-left: 17px;
  }
}
