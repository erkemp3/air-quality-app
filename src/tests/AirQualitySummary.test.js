import React from "react";
import { render } from "@testing-library/react";
import AirQualitySummary from "../components/AirQualitySummary";

describe("AirQualitySummary", () => {
  const validProps = {
    summaries: [
      {
        city: "Leeds",
        location: "Leeds Centre",

        measurements: [
          {
            lastUpdated: "2021-02-01",
            parameter: "pm10",
            value: 9,
          },
          {
            lastUpdated: "2021-03-01",
            parameter: "s02",
            value: 10,
          },
        ],
        onSubmit: () => {},
      },
      {
        city: "Manchester",
        location: "Manchester Centre",

        measurements: [
          {
            lastUpdated: "2021-04-01 19:00",
            parameter: "c02",
            value: 11,
          },
          {
            lastUpdated: "2021-05-01",
            parameter: "pm10",
            value: 15,
          },
        ],
        onSubmit: () => {},
      },
    ],
  };

  it("renders correctly", () => {
    const { asFragment } = render(
      <AirQualitySummary
        summaries={validProps.summaries}
        city={validProps.city}
        location={validProps.location}
        measurements={validProps.measurements}
        onSubmit={validProps.onSubmit}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders correct values for props", () => {
    const { getByText } = render(
      <AirQualitySummary
        summaries={validProps.summaries}
        city={validProps.city}
        location={validProps.location}
        measurements={validProps.measurements}
        onSubmit={validProps.onSubmit}
      />
    );

    expect(getByText("Leeds Centre")).toHaveClass("location");
    expect(getByText("in Manchester, United Kingdom")).toHaveClass("city");
  });
});
