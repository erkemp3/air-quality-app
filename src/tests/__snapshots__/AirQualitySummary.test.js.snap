// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AirQualitySummary renders correctly 1`] = `
<DocumentFragment>
  <div
    class="summaries"
  >
    <div
      class="summary"
    >
      <div
        class="button-div"
      >
        <button
          class="delete-button"
          type="submit"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
            />
          </svg>
        </button>
      </div>
      <p
        class="date"
      >
        LAST UPDATED: 01-02-2021 - 
      </p>
      <p
        class="location"
      >
        Leeds Centre
      </p>
      <p
        class="city"
      >
        in Leeds, United Kingdom
      </p>
      <p
        class="values"
      >
        Values: PM10: 9,  S02: 10
      </p>
    </div>
    <div
      class="summary"
    >
      <div
        class="button-div"
      >
        <button
          class="delete-button"
          type="submit"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
            />
          </svg>
        </button>
      </div>
      <p
        class="date"
      >
        LAST UPDATED: 01-04-2021 - 19:00
      </p>
      <p
        class="location"
      >
        Manchester Centre
      </p>
      <p
        class="city"
      >
        in Manchester, United Kingdom
      </p>
      <p
        class="values"
      >
        Values: C02: 11,  PM10: 15
      </p>
    </div>
  </div>
</DocumentFragment>
`;
